src/components/BaseTag.tsx

```diff
@@ -1,13 +1,24 @@
-import React from "react";
+import { MouseEventHandler } from "react";
 import styled from "styled-components";
-
 import DeleteButton from "./DeleteButton";
 
-const BaseTag = ({ tag, onClick, isEditable = false, className }) => {
+interface BaseTagProps {
+  tag: string;
+  onClick?: MouseEventHandler;
+  isEditable?: boolean;
+  className?: string;
+}
+
+const BaseTag = ({
+  tag,
+  onClick,
+  isEditable = false,
+  className,
+}: BaseTagProps) => {
   return (
     <TagEl className={className}>
-      <p>{isEditable ? { tag } : `#${tag}`}</p>
-      {isEditable && <DeleteButton key={tag} onClick={onClick} />}
+      <p>{isEditable ? tag : `#${tag}`}</p>
+      {isEditable && onClick && <DeleteButton onClick={onClick} />}
```

Comment:
```suggestion
      {isEditable && <DeleteButton onClick={onClick?.()} />}
```
이런것도 가능하답니다 :) 코드가 좀 더 깔끔해질수있겠네요!